crates/bevy_text/src/text.rs

```diff
@@ -279,6 +279,21 @@ pub enum FontSmoothing {
     // SubpixelAntiAliased,
 }
 
+/// Provides convenience methods for constructing text blocks.
+pub trait TextBuilderExt {
+    /// Adds a flat list of spans as children of the current entity.
+    fn with_spans<S: Component>(&mut self, spans: Vec<(S, TextStyle)>) -> &mut Self;
+}
+
+impl TextBuilderExt for EntityCommands<'_> {
+    fn with_spans<S: Component>(&mut self, mut spans: Vec<(S, TextStyle)>) -> &mut Self {
```

Comment:
This is better but it still leaves the awkwardness where you have to spawn the first element of a list separately as a `TextNEW` entity.
I was thinking of a builder that spawns the root `TextNew` entity as well.

Reply:
Added `spawn_text_block`